include/ignition/rendering/base/BaseLidarVisual.hh

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_RENDERING_BASELIDARVISUAL_HH_
+#define IGNITION_RENDERING_BASELIDARVISUAL_HH_
+
+#include "ignition/rendering/LidarVisual.hh"
+#include "ignition/rendering/base/BaseObject.hh"
+#include "ignition/rendering/base/BaseRenderTypes.hh"
+
+namespace ignition
+{
+  namespace rendering
+  {
+    inline namespace IGNITION_RENDERING_VERSION_NAMESPACE {
+    /// \brief Base implementation of a Lidar Visual
+    template <class T>
+    class BaseLidarVisual :
+      public virtual LidarVisual,
+      public virtual T
+    {
+      // Documentation inherited
+      protected: BaseLidarVisual();
+
+      // Documentation inherited
+      public: virtual ~BaseLidarVisual();
+
+      // Documentation inherited
+      public: virtual void PreRender() override;
+
+      // Documentation inherited
+      public: virtual void Destroy() override;
+
+      // Documentation inherited
+      public: virtual void ClearPoints() override;
+
+      // Documentation inherited
+      public: virtual void SetLidarMessage(std::vector<double> &msg);
+
+      // Documentation inherited
+      public: virtual void Update();
+
+      // Documentation inherited
+      public: virtual void Init();
+
+      // Documentation inherited
+      public: virtual void SetMinVerticalAngle(const double );
```

Comment:
please add the argument names, with undescore `_` -> _minVerticalAngle. Same with the methods bellow